src/main/java/seedu/address/logic/commands/UpdateCommand.java

```diff
@@ -119,7 +119,8 @@ private static Person createUpdatedPerson(Person personToUpdate, UpdatePersonDes
         Address updatedAddress = updatePersonDescriptor.getAddress().orElse(personToUpdate.getAddress());
         Description updatedDescription = updatePersonDescriptor.getDescription().orElse(personToUpdate
                 .getDescription());
-        NextOfKin updatedNextOfKin = updatePersonDescriptor.getNextOfKin().orElse(personToUpdate.getNextOfKin());
+        Optional<NextOfKin> updatedNextOfKin = Optional.ofNullable(updatePersonDescriptor.getNextOfKin()
```

Comment:
You can simplify this to `Optional<NextOfKin> updatedNextOfKin = updatePersonDescriptor.getNextOfKin().or(() -> personToUpdate.getNextOfKin())`